scripts/SANS/isis_reduction_steps.py

```diff
@@ -2115,40 +2125,44 @@ def setup_wksp(self, inputWS, inst, wavbining, trans_det_ids, reducer):
         tmp = mtd[tmpWS]
         # We perform a FlatBackground correction. We do this in two parts.
         # First we find the workspace indices which correspond to monitors
-        # and perform the correction on these indicies.
+        # and perform the correction on these indices.
         # Second we perform the correction on all indices which are not
         # monitors
-        for ws_index in range(tmp.getNumberHistograms()):
-            if tmp.getDetector(ws_index).isMonitor():
-                spectrum_number = tmp.getSpectrum(ws_index).getSpectrumNo()
-                back_start_mon, back_end_mon = inst.get_TOFs(spectrum_number)
-                if back_start_mon and back_end_mon:
-                    CalculateFlatBackground(
-                        InputWorkspace=tmpWS,
-                        OutputWorkspace=tmpWS,
-                        StartX=back_start_mon,
-                        EndX=back_end_mon,
-                        WorkspaceIndexList=ws_index,
-                        Mode='Mean')
-
-        back_start_roi, back_end_roi = inst.get_TOFs_for_ROI()
-        if back_start_roi and back_end_roi:
-            CalculateFlatBackground(
-                InputWorkspace=tmpWS,
-                OutputWorkspace=tmpWS,
-                StartX=back_start_roi,
-                EndX=back_end_roi,
-                WorkspaceIndexList=ws_index,
-                Mode='Mean',
-                SkipMonitors=True)
-
-        ConvertUnits(InputWorkspace=tmpWS, OutputWorkspace=tmpWS, Target="Wavelength")
-
-        if self.interpolate:
-            InterpolatingRebin(InputWorkspace=tmpWS, OutputWorkspace=tmpWS, Params=wavbining)
+        if tmp.getNumberHistograms() > 0:
+            ws_index = 0
```

Comment:
I see, the ws_index is used below, hence it requires initialization. Although I seem to have touched that area, I don't quite understand yet why this is required (nothing to do with this PR though).